src/coreclr/nativeaot/System.Private.CoreLib/src/System/RuntimeExceptionHelpers.cs

```diff
@@ -192,6 +192,29 @@ public struct ExceptionInformationArray
         }
 
         private static ulong s_crashingThreadId;
+        private static CrashInfo s_crashInfo;
+        private static int s_crashInfoPresent;
+
+        internal static CrashInfo SerializeCrashInfo(RhFailFastReason reason, string? message, Exception? exception)
+        {
+            int previousState = Interlocked.CompareExchange(ref s_crashInfoPresent, 1, 0);
+            if (previousState == 0)
+            {
+                s_crashInfoPresent = 1;
+
```

Comment:
```suggestion
            while (s_crashInfoPresent != 1)
            {
                 if (Interlocked.CompareExchange(ref s_crashInfoPresent, -1, 0))
                 {
                     // Some other thread is generating the crash info
                     Thread.Sleep(1);
                     continue;
                 }
                 
```
If there are multiple threads crashing (not unusual). we want the second thread to wait while the first thread is generating the crash info.

If we do not wait, the second thread will proceed to exit the process immediately, and we will likely end up with a crash without crash info.

Reply:
I am wondering whether we should have something similar in `AppContext.OnUnhandledException` in addition or instead of this. If there are multiple app threads crashing and each of them calls `AppContext.OnUnhandledException`, we will end up with multiple threads invoking the unhandled exception handlers. Unhandled exception handlers are not expected to be called from multiple thread or multiple times. 

cc @AaronRobinsonMSFT

Reply:
> I am wondering whether we should have something similar in AppContext.OnUnhandledException in addition or instead of this.

I agree. This is something we didn't really consider when it was added. This PR or some future one? I think it should be a .NET 10 item.

Reply:
Perhaps we can put the ExceptionHandling.RaiseAppDomainUnhandledExceptionEvent change to prevent multiple threads for invoking it into a subsequent PR, so that we can unblock this one for crash dump analysis/testing?

Reply:
Opened https://github.com/dotnet/runtime/issues/117840

Reply:
PR https://github.com/dotnet/runtime/pull/117844